### PR TITLE
[tests] remove beforeall from a bunch of tests

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Remove some `BeforeAll` usages to make  progress towards removing it completely. 20 more tests are now executed in parallel.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/vmi_configuration_test.go [hunk 1]
```diff
@@ -1661,8 +1661,7 @@ var _ = Describe("[sig-compute]Configurations", func() {
 			return updatedCPUName
 		}
 
-		// Collect capabilities once for all tests
-		tests.BeforeAll(func() {
+		BeforeEach(func() {
```
Comment (alonSadan) [inline tests/vmi_configuration_test.go:1664]: Can the two `beforeEach` (line 1664 and line 1669) be merged

File: tests/vmi_ignition_test.go [hunk 2]
```diff
@@ -35,31 +35,31 @@ import (
...
-			Expect(tests.WaitForSuccessfulVMIStart(obj)).ToNot(BeEmpty())
-		}
-	})
+		By("Waiting the VirtualMachineInstance start")
+		_, ok := obj.(*v1.VirtualMachineInstance)
+		Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
+		Expect(tests.WaitForSuccessfulVMIStart(obj)).ToNot(BeEmpty())
+	}
 
 	BeforeEach(func() {
```
Comment (alonSadan) [inline tests/vmi_ignition_test.go:62]: Can the two beforeEach (line 45 and line 62) be merged

**Review by alonSadan — COMMENTED** [review]
Thanks
some minor comments/questions added

File: tests/usbredir_test.go [hunk 3]
```diff
@@ -78,7 +78,7 @@ var _ = Describe("[Serial][crit:medium][vendor:cnv-qe@redhat.com][level:componen
 	})
 
 	Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component] A VirtualMachineInstance with usbredir support", func() {
-		tests.BeforeAll(func() {
+		BeforeEach(func() {
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/usbredir_test.go]: Yes. Done.

File: tests/vmi_cloudinit_test.go [hunk 4]
```diff
@@ -72,88 +72,85 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
-
-			out, err := tests.ExecuteCommandOnPod(virtClient, pod, "compute", cmdCheck)
-			Expect(err).NotTo(HaveOccurred())
-			size, err := strconv.Atoi(strings.Trim(out, "'"))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(size % 4096).To(Equal(0))
+	}
+
+	MountCloudInitNoCloud = mountCloudInitFunc("cidata")
+	MountCloudInitConfigDrive = mountCloudInitFunc("config-2")
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/vmi_cloudinit_test.go]: Done.

File: tests/vmi_configuration_test.go [hunk 5]
```diff
@@ -224,7 +224,7 @@ var _ = Describe("[sig-compute]Configurations", func() {
 			var availableNumberOfCPUs int
 			var vmi *v1.VirtualMachineInstance
 
-			tests.BeforeAll(func() {
+			BeforeEach(func() {
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/vmi_configuration_test.go:227]: Done.

File: tests/vmi_configuration_test.go (hunk 1 above)
Comment (rmohr) [reply tests/vmi_configuration_test.go:1662]: Done.

File: tests/vmi_ignition_test.go (hunk 2 above)
Comment (rmohr) [reply tests/vmi_ignition_test.go]: Done.

**Review by alonSadan — APPROVED** [review]
Thanks:)

File: tests/usbredir_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/usbredir_test.go:84]: Reverted it to two BeforeEach. The VMIs we want to start are differing.

**Comment by rmohr** [comment]
/cc @dhiller 

rebased this old PR which also helps running more tests in parallel. I think it is ready for a review.

**Comment by rmohr** [comment]
Fix for this failure https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6389/pull-kubevirt-e2e-k8s-1.23-sig-compute-cgroupsv2/1486632665489608704 is in #7119

**Review by dhiller — APPROVED** [review]
Thank you for this!

/approve

**Comment by dhiller** [comment]
> @rmohr: #6389 failed to apply on top of branch "release-0.49":
> 
> ```
> Applying: remove beforeall from a bunch of tests
> Using index info to reconstruct a base tree...
> M	tests/storage/storage.go
> M	tests/vmi_cloudinit_test.go
> M	tests/vmi_configuration_test.go
> Falling back to patching base and 3-way merge...
> Auto-merging tests/vmi_configuration_test.go
> Auto-merging tests/vmi_cloudinit_test.go
> CONFLICT (content): Merge conflict in tests/vmi_cloudinit_test.go
> Auto-merging tests/storage/storage.go
> error: Failed to merge in the changes.
> hint: Use 'git am --show-current-patch=diff' to see the failed patch
> Patch failed at 0001 remove beforeall from a bunch of tests
> When you have resolved this problem, run "git am --continue".
> If you prefer to skip this patch, run "git am --skip" instead.
> To restore the original branch and stop patching, run "git am --abort".
> ```

@rmohr FYI ^^

**Pull request merged**